Sources/Penny/Services/CacheService/S3CacheRepository.swift

```diff
@@ -0,0 +1,55 @@
+import SotoS3
+import Foundation
+
+struct S3CacheRepository {
+
+    let s3: S3
+    let logger: Logger
+    let bucket = "penny-caches"
+    let key = "caches.json"
+
+    init(awsClient: AWSClient, logger: Logger) {
+        self.s3 = S3(client: awsClient, region: .euwest1)
+        self.logger = logger
+    }
+
+    func get() async throws -> CacheStorage {
+        let request = S3.GetObjectRequest(bucket: bucket, key: key)
+        let response = try await s3.getObject(request, logger: logger)
+
+        if let buffer = response.body?.asByteBuffer(), buffer.readableBytes != 0 {
+            do {
+                return try JSONDecoder().decode(CacheStorage.self, from: buffer)
```

Comment:
Initializing `JSONDecoder`s is (sometimes) expensive; consider making the decoder an instance property (a static would run into threading issues).